src/Adapter/Builder/FtpAdapterDefinitionBuilder.php

```diff
@@ -66,12 +66,36 @@ protected function configureOptions(OptionsResolver $resolver)
 
         $resolver->setDefault('utf8', false);
         $resolver->setAllowedTypes('utf8', 'scalar');
+
+        $resolver->setDefault('transfer_mode', null);
+        $resolver->setAllowedTypes('transfer_mode', ['null', 'scalar']);
+        $resolver->setAllowedValues('transfer_mode', [null, FTP_ASCII, FTP_BINARY]);
+
+        $resolver->setDefault('system_type', null);
+        $resolver->setAllowedTypes('system_type', ['null', 'string']);
+        $resolver->setAllowedValues('system_type', [null, 'windows', 'unix']);
+
+        $resolver->setDefault('timestamps_on_unix_listings_enabled', false);
+        $resolver->setAllowedTypes('timestamps_on_unix_listings_enabled', 'bool');
+
+        $resolver->setDefault('recurse_manually', true);
+        $resolver->setAllowedTypes('recurse_manually', 'bool');        
     }
 
     protected function configureDefinition(Definition $definition, array $options)
     {
+        $options['transferMode'] = $options['transfer_mode'];
+        $options['systemType'] = $options['system_type'];
+        $options['timestampsOnUnixListingsEnabled'] = $options['timestamps_on_unix_listings_enabled'];
```

Comment:
Could we perhaps use the same name as the Flysystem option (ie in camel case by default)? If would avoid having to remap like this and would ease discoverability I think.

Reply:
I'd favor that too, but went with the existing code (when in Rome, do as the Romans do 🙂)

Reply:
Okay, it's as you prefer then, I don't mind either way.

Tests are failing though :)

Reply:
Shouldn't, anymore. Could you approve again please?